### ISSUE TITLE
Incorrect type inference for Set used in list comprehension

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1100]
```python
# pyright: strict

from typing import NamedTuple, Set

Target = NamedTuple("Target", [("categories", Set[str])])
all_targets = [Target(categories={"a", "b", "c"}), Target(categories={"b", "c", "d"})]


[category for target in all_targets for category in target.categories]
# error reported: Type of "category" is unknown
```

**Comment by erictraut** [comment]
This spurious error was reported only in Pylance, not in Pyright. That's because the semantic token provider in Pylance was causing nodes to be evaluated in an order that normally wouldn't occur, and that triggered a latent bug in the type evaluator.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.80, which I just published. It will also be in the next release of pylance.
